### PR TITLE
Remove deprecated Registration API

### PR DESCRIPTION
**Pull request by azdagron**
This API was replaced in 0.11.0 and deprecated in 0.12.0. We have kept it around for a little longer than usual to give folks time to migrate but it is time to remove it for 1.1.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -147,7 +147,6 @@ protos := \
 	proto/spire/common/common.proto \
 
 api-protos := \
```
Comment (MarcosDY) [inline Makefile:149]: api-protos will be empty we want to keep this var?

Comment (azdagron) [reply Makefile:149]: I stopped short of removing all of the infrastructure for generating API protos from the SPIRE Makefile. We haven't quite figured out how we want to stage new APIs and it's possible they live in SPIRE until they are ready to "graduate" and move to the API SDK.

**Review by amartinezfayo — COMMENTED** [review]
Looking good!
A quick search of "registration api" shows that there are some places that we still refer to it.
I've commented on one of them, that's the OIDC Discovery Provider documentation.
I also think that it would be good if we can update the migrating_registration_api_clients.md document to reflect that the API has been removed.

I believe that updating all the references to "registration api" would be pretty straightforward. What do you think?

**Comment by azdagron** [comment]
Oh good catch! I'll fix those up!

**Review by amartinezfayo — APPROVED** [review]
🎉

**Pull request merged**